pkg/backend/backend.go

```diff
@@ -161,7 +161,7 @@ type Backend interface {
 	RenameStack(ctx context.Context, stack Stack, newName tokens.QName) (StackReference, error)
 
 	// Preview shows what would be updated given the current workspace's contents.
-	Preview(ctx context.Context, stack Stack, op UpdateOperation) (engine.ResourceChanges, result.Result)
+	Preview(ctx context.Context, stack Stack, op UpdateOperation) (*deploy.Plan, engine.ResourceChanges, result.Result)
```

Comment:
This code structure is a tad confusing to me. Backend says it's an interface for `pluggable backend implementations of the Pulumi Cloud Service.` but it's not just storage functionality that's in the interface it's the engine functionality? And an implementation of this interface wraps the engine? Interesting.

Reply:
Urgh don't even get me started on this! The whole thing is a mad blob of everything and I need to gut a load of it to get project support in local backends sorted.